src/mame/drivers/mc10.cpp

```diff
@@ -26,123 +24,105 @@
 
 	uint8_t mc6847_videoram_r(offs_t offset);
-	TIMER_DEVICE_CALLBACK_MEMBER(alice32_scanline);
 
 	// device-level overrides
 	virtual void driver_start() override;
 	virtual void driver_reset() override;
 
 	required_device<m6803_cpu_device> m_maincpu;
 	required_device<ram_device> m_ram;
-	required_memory_bank m_bank1;
-	optional_memory_bank m_bank2;
+	optional_device<mc10cart_slot_device> m_mc10cart;
 
-	uint8_t *m_ram_base;
-	uint32_t m_ram_size;
 	uint8_t m_keyboard_strobe;
 	uint8_t m_port2;
 
 	uint8_t read_keyboard_strobe(bool single_line);
 
+	mc10cart_slot_device &mc10cart() { return *m_mc10cart; }
+
 private:
-	void alice32_mem(address_map &map);
-	void alice90_mem(address_map &map);
 	void mc10_mem(address_map &map);
 
 	optional_device<mc6847_base_device> m_mc6847;
-	optional_device<ef9345_device> m_ef9345;
 	required_device<dac_bit_interface> m_dac;
 	required_device<cassette_image_device> m_cassette;
 	required_device<rs232_port_device> m_rs232;
 	required_ioport_array<8> m_pb;
 };
 
-class mcx128_state : public mc10_state
+class alice32_state : public mc10_state
 {
 public:
-	mcx128_state(const machine_config &mconfig, device_type type, const char *tag)
+	alice32_state(const machine_config &mconfig, device_type type, const char *tag)
 		: mc10_state(mconfig, type, tag)
-		, m_mcx_ram(*this, "mcx_ram")
-		, m_mcx_cart_rom_base(*this, "cart")
-		, m_mcx_int_rom_base(*this, "maincpu")
-		, m_bank3(*this, "bank3")
-		, m_bank4r(*this, "bank4r")
-		, m_bank4w(*this, "bank4w")
-		, m_bank5r(*this, "bank5r")
-		, m_bank5w(*this, "bank5w")
-		, m_bank6r(*this, "bank6r")
-		, m_bank6w(*this, "bank6w")
-		{ }
-
-	void mcx128(machine_config &config);
-
-private:
-	uint8_t mcx128_bf00_r(offs_t offset);
-	void mcx128_bf00_w(offs_t offset, uint8_t data);
+		, m_ef9345(*this, "ef9345")
+	{}
 
-	void mcx128_mem(address_map &map);
-	void update_mcx128_banking();
+	void alice32(machine_config &config);
+	void alice32_bfff_w(uint8_t data);
 
+protected:
 	// device-level overrides
 	virtual void driver_start() override;
-	virtual void driver_reset() override;
 
-	required_device<ram_device> m_mcx_ram;
-	required_region_ptr<u8> m_mcx_cart_rom_base;
-	required_region_ptr<u8> m_mcx_int_rom_base;
-	required_memory_bank m_bank3;
-	required_memory_bank m_bank4r;
-	required_memory_bank m_bank4w;
-	required_memory_bank m_bank5r;
-	required_memory_bank m_bank5w;
-	required_memory_bank m_bank6r;
-	required_memory_bank m_bank6w;
-
-	uint8_t *m_mcx_ram_base;
-
-	uint8_t m_bank_control;
-	uint8_t m_map_control;
+	TIMER_DEVICE_CALLBACK_MEMBER(alice32_scanline);
+	required_device<ef9345_device> m_ef9345;
+
+private:
+	void alice32_mem(address_map &map);
+};
+
+
+class alice90_state : public alice32_state
+{
+public:
+	alice90_state(const machine_config &mconfig, device_type type, const char *tag)
+		: alice32_state(mconfig, type, tag)
+	{}
+
+	void alice90(machine_config &config);
+	uint8_t alice90_bfff_r();
+
+private:
+	void alice90_mem(address_map &map);
 };
```

Comment:
Is it worth having an extra class for this?  It doesn’t override any virtual member functions, it doesn’t add any required object finders that give benefits for validation.  Is there a good reason not to just roll it into the `alice32_state` class?